File: square-commons/src/main/java/com/raxdenstudios/square/interceptor/commons/handleextras/HandleExtrasActivityInterceptor.java
package com.raxdenstudios.square.interceptor.commons.handleextras;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.FragmentActivity;

import com.raxdenstudios.square.interceptor.ActivityInterceptor;

/**
 * Created by Ángel Gómez on 22/05/2015.
 */
public class HandleExtrasActivityInterceptor extends ActivityInterceptor<HandleExtrasInterceptorCallback> implements HandleExtrasInterceptor {

    public HandleExtrasActivityInterceptor(@NonNull FragmentActivity activity) {
        super(activity, null);
    }

    public HandleExtrasActivityInterceptor(@NonNull FragmentActivity activity, @NonNull HandleExtrasInterceptorCallback callback) {
        super(activity, callback);
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        Bundle extras = getActivity().getIntent() != null && getActivity().getIntent().getExtras() != null ? getActivity().getIntent().getExtras() : new Bundle();
        if (getCallback() != null) {
            getCallback().onHandleExtras(savedInstanceState, extras);
        }
    }

}

File: square-commons/src/main/java/com/raxdenstudios/square/interceptor/commons/handleextras/HandleExtrasActivityInterceptor.kt
package com.raxdenstudios.square.interceptor.commons.handleextras

import android.os.Bundle
import android.support.v4.app.FragmentActivity

import com.raxdenstudios.square.interceptor.ActivityInterceptor

/**
 * Created by Ángel Gómez on 22/05/2015.
 */
class HandleExtrasActivityInterceptor(
        activity: FragmentActivity,
        callback: HandleExtrasInterceptorCallback)
    : ActivityInterceptor<HandleExtrasInterceptorCallback>(activity, callback),
        HandleExtrasInterceptor {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        callback?.onHandleExtras(savedInstanceState, activity.intent.extras ?: Bundle())
    }

}
